### PR TITLE
SSCS-10295 use interpreter language if provided

### PR DESCRIPTION
**Pull request by pk2453**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/SSCS-10295

Change description
Interpreter requirements set in the adjournment notice is included in new hearing request

Does this PR introduce a breaking change? (check one with "x")

[x] Yes
[ ] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingPartiesMapping.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public static IndividualDetails getPartyIndividualDetails(Entity entity, Hearing
                 .firstName(HearingsPartiesMapping.getIndividualFirstName(entity))
                 .lastName(HearingsPartiesMapping.getIndividualLastName(entity))
                 .preferredHearingChannel(getIndividualPreferredHearingChannel(hearingSubtype, hearingOptions, null))
-                .interpreterLanguage(HearingsPartiesMapping.getIndividualInterpreterLanguage(hearingOptions, null, referenceData))
+            .interpreterLanguage(HearingsPartiesMapping.getIndividualInterpreterLanguage(hearingOptions, null, referenceData, null))
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingPartiesMapping.java:130]: indent

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java [hunk 2]
```diff
@@ -222,14 +221,18 @@ public static String getIndividualFullName(Entity entity) {
...
-    public static String getIndividualInterpreterLanguage(HearingOptions hearingOptions, OverrideFields overrideFields, ReferenceDataServiceHolder referenceData) throws InvalidMappingException {
+    public static String getIndividualInterpreterLanguage(HearingOptions hearingOptions, OverrideFields overrideFields, ReferenceDataServiceHolder referenceData, String adjournLanguage) throws InvalidMappingException {
 
         if (nonNull(overrideFields)
             && nonNull(overrideFields.getAppellantInterpreter())
             && nonNull(overrideFields.getAppellantInterpreter().getIsInterpreterWanted())) {
             return getOverrideInterpreterLanguage(overrideFields);
         }
 
+        if (nonNull(adjournLanguage)) {
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java:232]: Shouldn't this also check if adjournment flag is enabled && adjournmentInProgress? Bc if either of those are false then this should be ignored.

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java [hunk 3]
```diff
@@ -76,6 +56,48 @@ class HearingsPartiesMappingTest extends HearingsMappingBase {
...
+            null);
+        assertThat(partyDetails).isNotNull();
+        assertThat(partyDetails.getPartyType()).isNotNull();
+        assertThat(partyDetails.getPartyRole()).isNotNull();
+        assertThat(partyDetails.getIndividualDetails()).isNotNull();
+        assertThat(partyDetails.getOrganisationDetails()).isNull();
+        assertThat(partyDetails.getUnavailabilityDayOfWeek()).isEmpty();
+        assertThat(partyDetails.getUnavailabilityRanges()).isEmpty();
+
+        assertThat(hearingPartiesDetails.stream().filter(o -> DWP_ID.equalsIgnoreCase(o.getPartyID())).findFirst()).isNotPresent();
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:98]: I have been advised against using stream in assertions as they are hard to determine what is happening should they fail in regression. Try using "extracting" instead. I think there should be several examples from Lucas throughout the codebase.

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java [hunk 4]
```diff
@@ -295,12 +317,23 @@ void buildHearingPartiesDetailsJointParty(YesNo jointParty) throws InvalidMappin
...
 
         String individualInterpreterLanguage = HearingsPartiesMapping.getIndividualInterpreterLanguage(
-            null, null, referenceData
+            null, null, referenceData, null
         );
 
         assertThat(individualInterpreterLanguage).isNull();
     }
 
+    @DisplayName("When HearingOption is Null and and adjournLanguage is provided return empty string")
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:326]: and and

Comment (amoslv1) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:326]: Not clear what this test is testing. The description doesn't seem to match the assertion.

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java [hunk 5]
```diff
@@ -107,42 +105,42 @@ public static List<PartyDetails> buildHearingPartiesDetails(SscsCaseData caseDat
...
+                        otherParty.getHearingSubtype(), null, referenceDataServiceHolder, null));
             }
         }
 
         return partiesDetails;
     }
 
-    public static List<PartyDetails> buildHearingPartiesPartyDetails(Party party, String appellantId, ReferenceDataServiceHolder referenceData) throws InvalidMappingException {
-        return buildHearingPartiesPartyDetails(party, null, null, null, appellantId, null, referenceData);
+    public static List<PartyDetails> buildHearingPartiesPartyDetails(Party party, String appellantId, ReferenceDataServiceHolder referenceData, String adjournLanguage) throws InvalidMappingException {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java:115]: line is too long (check the rest of this file for them as theres a few it seems

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java [hunk 6]
```diff
@@ -335,8 +368,9 @@ void buildHearingPartiesPartyDetailsAppointee(String isAppointee, boolean expect
...
             hearingOptions,
             HearingSubtype.builder().build(),
-            PARTY_ID,
-            null, referenceData
+            null,
+            referenceData,
+            null
         );
 
         assertThat(partiesDetails.stream().filter(o -> PARTY_ID.substring(0,15).equalsIgnoreCase(o.getPartyID())).findFirst()).isPresent();
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:376]: I see that this stream pattern is throughout the test class. Perhaps you can find a way to use extracting and replace all of them?

**Review by nickhill111 — DISMISSED** [review]
approved but 1 minor comment

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java [hunk 7]
```diff
@@ -8,11 +8,7 @@
 import org.junit.jupiter.params.provider.NullSource;
 import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Adjournment;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
-import uk.gov.hmcts.reform.sscs.ccd.domain.HearingOptions;
-import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
-import uk.gov.hmcts.reform.sscs.ccd.domain.YesNo;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java:11]: Revert this * import to individual specified

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java [hunk 8]
```diff
@@ -360,17 +354,20 @@ void urgentCase(String urgentCase) {
...
-    @ParameterizedTest
-    @ValueSource(strings = {"spanish", "french"})
-    void adjournCaseInterpreterLanguage(String interpreterLanguage) {
+    @DisplayName("adjournCaseInterpreterLanguage should return party flag mapping if interpreter required")
+    @Test
+    void adjournCaseInterpreterLanguage() {
         SscsCaseData caseData = SscsCaseData.builder()
             .adjournment(Adjournment.builder()
-                .interpreterLanguage(interpreterLanguage)
+                    .interpreterRequired(YES)
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java:362]: This should also have adjournmentInProgress(YES) in order to be in effect.

Likewise, a test is required to show that when it is NO, this does not trigger.

**Review by amoslv1 — CHANGES_REQUESTED** [review]
Mainly some questions around checking adjournmentInProgress == YES and feature flag == true

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java (hunk 3 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java:119]: This is already there at many places (just copy/paste from existing one) in this file so should go as tech deb.,

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java (hunk 7 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java:11]: Done.

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java (hunk 8 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java:363]: Test added for NO

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingPartiesMapping.java (hunk 1 above)
Comment (pk2453) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingPartiesMapping.java]: fixed

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java (hunk 4 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMappingTest.java]: fixed

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java (hunk 5 above)
Comment (pk2453) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java]: fixed

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java (hunk 7 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java:11]: fixed

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java (hunk 8 above)
Comment (pk2453) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java:363]: test added for No

**Review by amoslv1 — APPROVED** [review]

**Review by amoslv1 — DISMISSED** [review]
Subject to Lucas confirming that feature flag / adjournmentInProgress checks are not required

**Review by jarekPierchala — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Pull request merged**